pkg/lib/sidecars/test/features/update.feature

```diff
@@ -20,10 +20,46 @@ Feature: Istio upgrade
     And all required resources are deleted
   
   Scenario: Standard reconciliation
-    Given there is a cluster with Istio "1.14.4", default injection == "false" and CNI enabled == "false"
+    Given there is a cluster with Istio "1.14.4", default injection == "false"
     And there are Pods with Istio "1.14.4" sidecar
     When a restart happens with target Istio "1.14.4"
     Then no resource that is not supposed to be restarted is restarted
     And no resource that is not supposed to be deleted is deleted
     And all required resources are restarted
+    And all required resources are deleted
+
+  Scenario: Updating CPU requests Proxy configuration
+    Given there is a cluster with Istio "1.14.4", default injection == "true"
+    And there are Pods with Istio "1.14.4" sidecar and resource "requests" cpu "10m" and memory "100Mi"
+    When a restart happens with target Istio "1.14.4" and sidecar resource "requests" set to cpu "20m" and memory "100Mi"
+    Then no resource that is not supposed to be restarted is restarted
+    And no resource that is not supposed to be deleted is deleted
+    And all required resources are restarted
+    And all required resources are deleted
+
```

Comment:
Should we check the resources afterwards?
```suggestion
And there are Pods with Istio "1.14.4" sidecar and resource "requests" cpu "20m" and memory "100Mi"

```

Reply:
As discussed, those are the mocked sidecar tests that don't updated the resources, so we don't want to verify it here and it's already verified in the integration tests.